viewing_party/main.py

```diff
@@ -0,0 +1,161 @@
+
+    # Create a new movie
+    new_movie = {"title": movie_title, "genre": genre, "rating": rating}
+
+    if new_movie["title"] == None or new_movie["genre"] == None or new_movie["rating"] == None:
+        return None
+    return new_movie
+
+def add_to_watched(user_data, movie):
+
+    # Add movie in the list of movies that the user has already watched
+    user_data = {
+        "watched": [movie]
+    }
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+
+    # Add movie in the user's watchlist
+    user_data = {
+        "watchlist": [movie]
+    }
+    return user_data
+
+def watch_movie(user_data, title):
+
+    # Check if title is in a movie in the user's watchlist 
+    for movie in user_data["watchlist"]:
+        if movie["title"] == title:
+        # Add movie in the list of watched movie and remove it from the watchlist
+            user_data["watched"].append(movie)
+            user_data["watchlist"].remove(movie)
+            return user_data
+    return user_data
+
+def get_watched_avg_rating(user_data):
+
+    # Setup
+    movie_rating = []
+    avg_rating = None
+
+    # Check if there are movies in the list of watch movies
+    if len(user_data["watched"]) == 0:
+        avg_rating = 0.0
+        return avg_rating 
+    # Check if there are any movie in the list of watch movies
+    else:
+        for movie in user_data["watched"]:
+        # Add movie in a new list
+            movie_rating.append(movie["rating"])
+        # Calculate the average rating
+        avg_rating = sum(movie_rating)/len(movie_rating)
+        return avg_rating
+
+def get_most_watched_genre(user_data):
+
+    # Setup
+    l_genre =  {}
+    
+    # Check if movie genre is in the l_genre dictionary and if not, we add the genre as a key with a value of 1 
+    # Otherwise, increment the value by 1
+    for movie in user_data["watched"]:
+        if movie["genre"] not in l_genre:
+            l_genre[movie["genre"]] = 1
+        else:
+            l_genre[movie["genre"]] += 1
+    # Setup
+    number= 0
+    popular_genre= None
+
+    # Find genre with the highest value in the l_genre dictionary
+    for key in l_genre:
+    # Compare genre value with number and update number if genre value is highest than number
+        if l_genre[key] > number:
+            number = l_genre[key]
+            popular_genre = key
+    return popular_genre
+
+def get_unique_watched(user_data):
+    
+    #Setup
+    movies_watched = []
+    unique_movies = []
+    user_friends = user_data["friends"]
+
+    # Add movies that friends have watched in the movies_watched list
+    for friend in user_friends:
+        for value in friend.values():
+            movies_watched.extend(value)
+
+    # Check if movie that user has wacthed is in the movies_watched list
+    for movie in user_data["watched"]:
+        if movie in movies_watched:
+            continue
+        unique_movies.append(movie)
+    print(movies_watched)
```

Comment:
oop don't forget to get rid of any print statements that are simply for debugging before you submit a final project